src/application/models/v3/Appointments_model_v3.php

```diff
@@ -18,32 +18,44 @@
  */
 class Appointments_Model_V3 extends Appointments_Model {
 
-    public function getAppointmentWithIdUserIntegrated($id_user_integrated) {
+    public function getAppointmentsWithCondition($id_user_integrated, 
+                                                $id_service_integrated, 
+                                                $id_patient_integrated,
+                                                $startDate,
+                                                $endDate,
+                                                $page,
+                                                $size,
+                                                $sort) {     
+        $resultSet['total'] = $this->countAppointmentsByCondition($id_user_integrated, $id_service_integrated, $id_patient_integrated, $startDate, $endDate);
+        $arrayParams = $this->initStoredProcedureParams($id_user_integrated, $id_service_integrated, $id_patient_integrated, $startDate, $endDate);               
+        $appointmentsQuery = 'CALL getAppointmentsByConditionAndPaging(?, ?, ?, ?, ?, ?, ?, ?)';
+        array_push($arrayParams, $page, $size, $sort);
+        $query = $this->db->query($appointmentsQuery, $arrayParams);
+        $resultSet['appointments'] = $query->result_array();
+        $this->releaseStoredProcedureQuery($query);    
+        return $resultSet; 
+    }
 
-        return $this->db->select('*')->from('ea_appointments')
-        ->join('integrated_users_patients', 'ea_appointments.id_users_customer = integrated_users_patients.id_patients')
-        ->where('integrated_users_patients.id_user_integrated', $id_user_integrated)->get()->result_array();
+    private function countAppointmentsByCondition($id_user_integrated, $id_service_integrated, $id_patient_integrated, $startDate, $endDate){
+        $arrayParams = $this->initStoredProcedureParams($id_user_integrated, $id_service_integrated, $id_patient_integrated, $startDate, $endDate);
+        $totalQuery = 'CALL countAppointmentsByCondition(?, ?, ?, ?, ?)';
```

Comment:
extract the statement to a constant class

Reply:
updated